tutorials/image/cifar10_estimator/cifar10_main.py

```diff
@@ -220,147 +145,172 @@ def __call__(self, op):
     return device_name
 
 
-def _create_device_setter(is_cpu_ps, worker, num_gpus):
+def _create_device_setter(avg_on_gpu, worker, num_gpus):
```

Comment:
Yes, in fact, I think this bit of code could be replaced by the following, which I believe is equivalent! 
```
placement_strat = tf.contrib.training.GreedyLoadBalancingStrategy(
  num_ps, load_fn=tf.contrib.training.byte_size_load_fn)
with tf.device(tf.train.replica_device_setter(worker_device=worker,
                                                                    ps_strategy=placement_strat)):
   # tower
```
Which I think should be almost identical (your device_fn uses number of elements rather than byte size, but if they are all the same type those should be equivalent).